### PR TITLE
remove pvsystem.systemdef

### LINKED ISSUE #965: remove pvsystem.systemdef function
Does anyone use this function? Does anyone have use case for it? I'm inclined to remove it without deprecation in 0.8.

Another option is to move it to `pvlib.iotools.tmy`.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #965
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - ~Tests added~
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

I was surprised by the very large number of removed lines in the `pvsystem.ipynb` file. Maybe relevant: I edited and reran the file using the VSCode's jupyter extension. First time doing that, but it looks ok to me. Really dislike keeping those in the main repo.

### CODE REVIEW

File: docs/sphinx/source/whatsnew/v0.8.0.rst [hunk 1]
```diff
@@ -12,6 +12,11 @@ API Changes with Deprecations
    - :py:func:`pvlib.pvsystem.adrinverter` is now :py:func:`pvlib.inverter.adr`
 * Argument ``ac_model`` for :py:class:`pvlib.modelchain.ModelChain` now accepts
   ``'sandia'``, ``'pvwatts'`` and ``'adr'`` for the inverter models.  (:pull:`886`)
+* Removed ``systemdef`` function from ``pvsystem.py``. This function was not
+  used in pvlib and its output was not directly compatible with any pvlib
+  function. See :py:func:`pvlib.iotools.read_tmy2`,
+  :py:func:`pvlib.iotools.read_tmy3`, :py:meth:`pvlib.Location.from_tmy`, and
```
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.8.0.rst:18]:
```suggestion
  :py:func:`pvlib.iotools.read_tmy3`, :py:meth:`pvlib.location.Location.from_tmy`, and
```

File: pvlib/tests/test_pvsystem.py [hunk 2]
```diff
@@ -16,64 +16,7 @@
...
-                'altitude': 10,
-                'latitude': 37.8,
-                'longitude': -122.3,
-                'name': 'Oakland',
-                'strings_per_inverter': 5,
-                'modules_per_string': 5,
-                'surface_azimuth': 0,
-                'surface_tilt': 5}
-    assert expected == pvsystem.systemdef(meta, 5, 0, .1, 5, 5)
+    needs_numpy_1_10, requires_scipy, fail_on_pvlib_version)
```
Comment (kanderso-nrel) [inline pvlib/tests/test_pvsystem.py:19]: Tiny nit, this import might fit on one line now

File: docs/sphinx/source/whatsnew/v0.8.0.rst [hunk 3]
```diff
@@ -12,6 +12,11 @@ API Changes with Deprecations
...
+  :py:class:`pvlib.pvsystem.LocalizedPVSystem` for alternatives. (:issue:`965`)
```
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.8.0.rst:19]: Usually the PR is linked as well, though it's easy to get to from the issue page so I don't care about being a stickler about this.

Comment (kanderso-nrel) [reply docs/sphinx/source/whatsnew/v0.8.0.rst:19]: Also, this is in the "API Changes with Deprecations" section, but I don't think this was deprecated was it?

**Comment by wholmgren** [comment]
> I took a look at the source code of the new notebook just for curiosity. The big difference in line count is mostly because table HTML is all in one line now instead of one cell per line. Interestingly, some of the images are encoded as SVG instead of base64 PNG -- very cool. I don't see an issue with those differences except I suppose there will be an inversely large diff next time someone edits the notebooks without using the VSCode plugin.

Thanks that's good to know. I wonder if it's less about vscode and more about jupyterlab vs notebook or newer versions of the dependencies. 

> Usually the PR is linked as well, though it's easy to get to from the issue page so I don't care about being a stickler about this.

Agree that a PR link is a nice to have rather than a necessity. 

> Also, this is in the "API Changes with Deprecations" section, but I don't think this was deprecated was it?

Thanks for catching this. #1009.

**Pull request merged**